scripts/convert_mapping_to_schema.py

```diff
@@ -0,0 +1,117 @@
+import argparse
+from collections import defaultdict
+import os
+import csv
+import json
+from typing import NamedTuple, Any, Iterable
+
+
+class JadeColumn(NamedTuple):
+    column_name: str
+    column_type: str
+
+
+TDR_MAPPING_FILE_COLUMN_NAME_ID = "TDR_Column_Name"
+TDR_MAPPING_FILE_JADE_FRAGMENT = "Jade Fragment?"
+TDR_MAPPING_COLUMN_TYPE_ID = "TDR_Variable_Type"
+TDR_FRAGMENT_GENERAL = "GENERAL"
+TDR_RC_TO_COLUMN_TYPES = {
+    "BOOL": "boolean",
+    "INT": "integer",
+    "STRING": "string"
+}
+
+
+def parse_csv(mapping_csv: str) -> dict[str, list[JadeColumn]]:
+    fragments = defaultdict(list[JadeColumn])
+
+    with open(mapping_csv) as f:
+        csv_reader = csv.DictReader(f)
+        for row in csv_reader:
+            raw_type_id = row[TDR_MAPPING_COLUMN_TYPE_ID]
+            raw_column_name_id = row[TDR_MAPPING_FILE_COLUMN_NAME_ID]
+
+            if not raw_type_id and not row[TDR_MAPPING_FILE_JADE_FRAGMENT]:
+                continue
+
+            if raw_type_id not in TDR_RC_TO_COLUMN_TYPES.keys():
+                raise Exception(
+                    f"Invalid column, field name = {row[TDR_MAPPING_FILE_COLUMN_NAME_ID]}, type = {row[TDR_MAPPING_COLUMN_TYPE_ID]}")
+
+            jade_field = JadeColumn(
+                column_name=raw_column_name_id,
+                column_type=TDR_RC_TO_COLUMN_TYPES[raw_type_id]
+            )
+
+            fragment = TDR_FRAGMENT_GENERAL
+            if row[TDR_MAPPING_FILE_JADE_FRAGMENT]:
+                fragment = row[TDR_MAPPING_FILE_JADE_FRAGMENT]
+
+            fragments[fragment].append(jade_field)
+
+    return fragments
+
+
+def render_schema_fragment(fragment: str, fields: list[JadeColumn], table_name: str) -> dict[str, Any]:
+    name = f"{table_name}_{fragment}"
+    if fragment == TDR_FRAGMENT_GENERAL:
+        name = table_name
+
+    columns = []
+    for field in fields:
+        columns.append({
+            "name": field.column_name,
+            "datatype": field.column_type
+        })
+
+    return {
+        "name": name,
+        "columns": columns
+    }
+
+
+def render_general_fragment(general_fragment, table_fragments: Iterable[str], table_name: str):
+    dog_id_field = {
+      "name": "dog_id",
+      "datatype": "integer",
+      "type": "primary_key",
+      "links": [
+        {
+          "table_name": "hles_dog",
+          "column_name": "dog_id"
+        }
+      ]
+    }
+    schema = render_schema_fragment(TDR_FRAGMENT_GENERAL, general_fragment, table_name)
+    schema["columns"].insert(0,dog_id_field)
+    schema["table_fragments"] = list(table_fragments)
+
+    return schema
+
+
+def render_schemas(fragments: dict[str, list[JadeColumn]], table_name: str):
+    general_fragment = fragments.pop(TDR_FRAGMENT_GENERAL)
```

Comment:
🔧 I think this should handle the edge case where there is no general fragment ( i.e., everything is bucketed) with a default empty dict:

```suggestion
    general_fragment = fragments.pop(TDR_FRAGMENT_GENERAL, {})
```